pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1104,41 +1104,25 @@ func Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInsta
 		}
 
 		if vmi.Spec.Domain.Firmware.Bootloader != nil && vmi.Spec.Domain.Firmware.Bootloader.EFI != nil {
-			// The location of uefi boot loader on ARM64 is different from that on x86 and ppc64le
-			efiCode := ""
-			efiVars := ""
-			if isARM64(c.Architecture) {
-				efiCode = EFICodeAARCH64
-				efiVars = EFIVarsAARCH64
-			} else {
-				efiCode = EFICode
-				efiVars = EFIVars
-			}
+			secureBoot := vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot == nil || *vmi.Spec.Domain.Firmware.Bootloader.EFI.SecureBoot
+			efiCode, _ := detectEFICodeRom(c, secureBoot)
```

Comment:
@davidvossel @acardace information like  this has to be passed in through the converter context.

The conversion is split into two phases:

 1. gathering all information and add it to the converter context
 2. making use of that information in the converter context
 
 Can you please do a follow-up PR and fix that?

Reply:
Yes I can ideed, thanks for taking a look!